blockchain/types/tx_internal_data_access_list.go

```diff
@@ -0,0 +1,494 @@
+	return accountkey.RoleTransaction
+}
+
+func (t *TxInternalDataAccessList) ChainId() *big.Int {
+	if t.ChainID != nil {
+		return t.ChainID
+	}
+	return deriveChainId(t.V)
+}
+
+func (t *TxInternalDataAccessList) GetAccountNonce() uint64 {
+	return t.AccountNonce
+}
+
+func (t *TxInternalDataAccessList) GetPrice() *big.Int {
+	return new(big.Int).Set(t.Price)
+}
+
+func (t *TxInternalDataAccessList) GetGasLimit() uint64 {
+	return t.GasLimit
+}
+
+func (t *TxInternalDataAccessList) GetRecipient() *common.Address {
+	return t.Recipient
+}
+
+func (t *TxInternalDataAccessList) GetAmount() *big.Int {
+	return new(big.Int).Set(t.Amount)
+}
+
+func (t *TxInternalDataAccessList) GetHash() *common.Hash {
+	return t.Hash
+}
+
+func (t *TxInternalDataAccessList) GetPayload() []byte {
+	return t.Payload
+}
+
+func (t *TxInternalDataAccessList) GetAccessList() AccessList {
+	return t.AccessList
+}
+
+func (t *TxInternalDataAccessList) SetHash(hash *common.Hash) {
+	t.Hash = hash
+}
+
+func (t *TxInternalDataAccessList) SetSignature(signatures TxSignatures) {
+	if len(signatures) != 1 {
+		logger.Crit("AccessListTransaction receives a single signature only!")
+	}
+
+	t.V = signatures[0].V
+	t.R = signatures[0].R
+	t.S = signatures[0].S
+}
+
+func (t *TxInternalDataAccessList) RawSignatureValues() TxSignatures {
+	return TxSignatures{&TxSignature{t.V, t.R, t.S}}
+}
+
+func (t *TxInternalDataAccessList) ValidateSignature() bool {
+	v := byte(t.V.Uint64())
+	return crypto.ValidateSignatureValues(v, t.R, t.S, false)
+}
+
+func (t *TxInternalDataAccessList) RecoverAddress(txhash common.Hash, homestead bool, vfunc func(*big.Int) *big.Int) (common.Address, error) {
+	V := vfunc(t.V)
+	return recoverPlain(txhash, t.R, t.S, V, homestead)
+}
+
+func (t *TxInternalDataAccessList) RecoverPubkey(txhash common.Hash, homestead bool, vfunc func(*big.Int) *big.Int) ([]*ecdsa.PublicKey, error) {
+	V := vfunc(t.V)
+
+	pk, err := recoverPlainPubkey(txhash, t.R, t.S, V, homestead)
+	if err != nil {
+		return nil, err
+	}
+
+	return []*ecdsa.PublicKey{pk}, nil
+}
+
+func (t *TxInternalDataAccessList) IntrinsicGas(currentBlockNumber uint64) (uint64, error) {
+	return IntrinsicGas(t.Payload, t.AccessList, t.Recipient == nil, *fork.Rules(big.NewInt(int64(currentBlockNumber))))
+}
+
+func (t *TxInternalDataAccessList) setSignatureValues(chainID, v, r, s *big.Int) {
+	t.ChainID, t.V, t.R, t.S = chainID, v, r, s
+}
+
+func (t *TxInternalDataAccessList) SerializeForSign() []interface{} {
+	return []interface{}{
+		t.AccountNonce,
+		t.Price,
+		t.GasLimit,
+		t.Recipient,
+		t.Amount,
+		t.Payload,
+		t.AccessList,
+	}
+}
+
+func (t *TxInternalDataAccessList) SenderTxHash() common.Hash {
+	return prefixedRlpHash(byte(t.Type()), []interface{}{
+		t.ChainID,
+		t.AccountNonce,
+		t.Price,
+		t.GasLimit,
+		t.Recipient,
+		t.Amount,
+		t.Payload,
+		t.AccessList,
+		t.V,
+		t.R,
+		t.S,
+	})
+}
+
+func (t *TxInternalDataAccessList) IsLegacyTransaction() bool {
+	return true
```

Comment:
Is this true?

Reply:
Yes. This function is used to distinguish whether a transaction has no sender field or not.

Reply:
Umm the function name sounds so ambiguous to use that way.
What is your thinking of using `IsEthTypedTransaction` function to determine this tx types has sender or not? Is this possible?